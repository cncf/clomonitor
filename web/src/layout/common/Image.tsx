import { isUndefined } from 'lodash';
import isNull from 'lodash/isNull';
import { useRef, useState } from 'react';
import { MdOutlineImageNotSupported } from 'react-icons/md';

interface Props {
  url?: string | null;
  alt: string;
  className?: string;
}

const Image = (props: Props) => {
  const image = useRef<HTMLImageElement | null>(null);
  const [error, setError] = useState(false);

  return (
    <>
      {error || isNull(props.url) || isUndefined(props.url) ? (
        <MdOutlineImageNotSupported data-testid="img-placeholder" />
      ) : (
        <img
          ref={image}
          alt={props.alt}
          src={props.url}
          className={props.className}
          onError={() => setError(true)}
          aria-hidden="true"
        />
      )}
    </>
  );
};

export default Image;
