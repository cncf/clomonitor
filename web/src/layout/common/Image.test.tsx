import { render, screen } from '@testing-library/react';

import Image from './Image';

const defaultProps = {
  url: 'http://img.url',
  alt: 'image',
};

describe('Image', () => {
  afterEach(() => {
    jest.resetAllMocks();
  });

  it('creates snapshot', () => {
    const { asFragment } = render(<Image {...defaultProps} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders proper content', () => {
    render(<Image {...defaultProps} />);

    const img = screen.getByAltText('image');
    expect(img).toBeInTheDocument();
    expect(img).toHaveAttribute('src', 'http://img.url');
  });

  it('renders placeholder', () => {
    render(<Image {...defaultProps} url={null} />);

    expect(screen.queryByAltText('image')).toBeNull();
    expect(screen.getByTestId('img-placeholder')).toBeInTheDocument();
  });
});
