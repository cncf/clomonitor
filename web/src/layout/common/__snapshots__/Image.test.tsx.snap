// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image creates snapshot 1`] = `
<DocumentFragment>
  <img
    alt="image"
    aria-hidden="true"
    src="http://img.url"
  />
</DocumentFragment>
`;
